### DISCUSSION TITLE
Can I use GRNBoost output to SCENIC R pipline?

### DISCUSSION THREAD

**Discussion by lh12565** [aertslab/SCENIC#262] [R]
Hi,

I use GRNBoost to infer a gene regulatory network. I want to use the output to further analysis in R as follows:
```
grnres<-read.table('grnboost_output.csv',stringsAsFactors=F,sep="\t")
saveRDS(grnres,"int/1.4_GENIE3_linkList.Rds")
runSCENIC_1_coexNetwork2modules(scenicOptions)
runSCENIC_2_createRegulons(scenicOptions) 
runSCENIC_3_scoreCells(scenicOptions, logMat)
```
But their column is different, so I changed the column "TF  Target important"  to "TF Target weight" as #40 said. I don't know is it correct to do this. And  the GENIE3 weights are <1 while the GRNBoost importance range from 0-100+, so does this affect the results? 
I also see the pyscenic workflow, but I didn't find the motif database and I'm not familiar with python, so I use R to do the next step. Thanks!

**Comment by morganee261** [comment]
Hi @lh12565,
I am trying to import my pyscenic outputs into R to continue the analysis and am running into issues. Any luck with that?

thanks,
Morgane

**Comment by lh12565** [comment]
Hi @morganee261, I didn't use the pyscenic, and I only used grnboost (https://arboreto.readthedocs.io/en/latest/) to infer a gene regulatory network. What is your problem when you use the pyscenic?

**Comment by millersan** [comment]
I use this result like you show , I had run genie3 before in bulk data , the result is more likely grnboost . I succeed to get the final result .

**Comment by luoboqingcai81** [comment]
i meet the same issue,  the GENIE3 used in SCENIC is so slow.  I used the GRNBoost in python, but the result is different from the GENIE3 in SCENIC

**Comment by Chenmengpin** [comment]
> Hi,
> 
> I use GRNBoost to infer a gene regulatory network. I want to use the output to further analysis in R as follows:
> 
> ```
> grnres<-read.table('grnboost_output.csv',stringsAsFactors=F,sep="\t")
> saveRDS(grnres,"int/1.4_GENIE3_linkList.Rds")
> runSCENIC_1_coexNetwork2modules(scenicOptions)
> runSCENIC_2_createRegulons(scenicOptions) 
> runSCENIC_3_scoreCells(scenicOptions, logMat)
> ```
> 
> But their column is different, so I changed the column "TF Target important" to "TF Target weight" as #40 said. I don't know is it correct to do this. And the GENIE3 weights are <1 while the GRNBoost importance range from 0-100+, so does this affect the results?
> I also see the pyscenic workflow, but I didn't find the motif database and I'm not familiar with python, so I use R to do the next step. Thanks!

Hi, I meet the same issue like you described, did you get any answers? Thanks!

**Comment by linxy29** [comment]
I also have the same problem and I'm seeking a solution.

**Comment by enoraFremy** [comment]
I use Scenic in R, but the step with GENIE3 is too slow. So, I used pyscenic grn instead. But runSCENIC needs a file a little different from the output of GRNBoost: 
(1) importArboreto(\<output GRNBoost adj.tsv\>)
(2) change the column from "TF Target Importance ImportanceNorm" to "TF Target ~~Importance~~ weight"
(3) create the folder int/ and rename the tab "int/1.4_GENIE3_linkList.Rds". That file works as input for runSCENIC (also need to initialize  scenicOptions)

```
grnboost_linklist <- importArboreto("adj.tsv")
grnboost_linklist <- grnboost_linklist[c(1,2,4)]
names(grnboost_linklist) <- c("TF", "Target", "weight")
saveRDS(grnboost_linklist, "int/1.4_GENIE3_linkList.Rds")

scenicOptions <- initializeScenic(org="hgnc", dbDir="/path/to/cisTarget_databases", nCores=10)
scenicOptions@inputDatasetInfo$cellInfo <- "int/cellInfo.Rds"
scenicOptions@settings$verbose <- TRUE
scenicOptions@settings$nCores <- 10
scenicOptions@settings$seed <- 123
scenicOptions@settings$dbs <- scenicOptions@settings$dbs["10kb"]
saveRDS(scenicOptions, file="int/scenicOptions.Rds")
```
